src/components/TransactionStatus.tsx

```diff
@@ -101,29 +120,40 @@ function TransactionStatus({ transaction, onComplete }: Props) {
       return;
     }
 
-    const stepEvaluator = (transactionValue: string | number | null, chainValue: string | number | null) => {
+    const stepEvaluator = (transactionValue: string | number | null, chainValue: string | number | null): boolean => {
       if (!transactionValue || !chainValue) return false;
 
       const bnChainValue = new BN(chainValue);
       const bnTransactionValue = new BN(transactionValue);
       return bnChainValue.gt(bnTransactionValue);
     };
 
-    const completionStatus = (status: boolean) =>
-      transaction.block && status ? TransactionStatusEnum.COMPLETED : TransactionStatusEnum.IN_PROGRESS;
+    const completionStatus = (status: boolean = false): TransactionStatusEnum => {
+      if (transaction.id === 0) {
+        return TransactionStatusEnum.NOT_STARTED;
+      }
+      if (!status) {
+        return transaction.block ? TransactionStatusEnum.COMPLETED : TransactionStatusEnum.IN_PROGRESS;
+      }
+      return transaction.block && status ? TransactionStatusEnum.COMPLETED : TransactionStatusEnum.IN_PROGRESS;
```

Comment:
```suggestion
      if (status) {
        return transaction.block ? TransactionStatusEnum.COMPLETED : TransactionStatusEnum.IN_PROGRESS;
      }
      return TransactionStatusEnum.IN_PROGRESS;
```

Now I've noticed the conditions are a bit weird, I think that's the correct behavior.

Reply:
What you propose is not correct. There is only 1 case that there is no status passed to the function (1st step):
```
{
        chainType: sourceChain,
        label: 'Include message in block',
        onChain: transaction.block,
        status: completionStatus()
      }
   ```
   On this case we may have the same if/else but we do not check for status(if provided) compared to all rest of steps - and still we want the TransactionStatusEnum.COMPLETED to return

Reply:
But what you have now has no sense at all - `status` is known to be `true`, so you can remove `&& status` and then you end up with exactly the same thing in both `if` and outside conditions so it means `status` does not matter at all.

Reply:
It matters on the first step where the call is completionStatus() [meaning status is false (init from function param)].
We need to have this step for the (non given status) step 1, but also for the rest of the steps where status SHOULD be given - but in case it is not (status = false), then the transactionStatus is in progress.
If I alter the code the way you propose, then step 1 will ALWAYS be IN_PROGRESS and will never be completed (as no parameter, ever, is passed to the completionStatus() function.

Reply:
Please take one more look at this code:
```
if (!status) {
        return transaction.block ? TransactionStatusEnum.COMPLETED : TransactionStatusEnum.IN_PROGRESS;
      }
      /// status == true - cause another case was handled above
      return transaction.block && status ? TransactionStatusEnum.COMPLETED : TransactionStatusEnum.IN_PROGRESS;
```

So it can be simplified to:
```
if (!status) {
        return transaction.block ? TransactionStatusEnum.COMPLETED : TransactionStatusEnum.IN_PROGRESS;
      }
      return transaction.block ? TransactionStatusEnum.COMPLETED : TransactionStatusEnum.IN_PROGRESS;
```

Which in turn can be simplified to:
```
 return transaction.block ? TransactionStatusEnum.COMPLETED : TransactionStatusEnum.IN_PROGRESS;
```

hence `status` does not matter and that's why I assume there might be a flaw in your logic. I'm not arguing that my solution is the correct one, but I do argue that current code is incorrect.

Reply:
Code you propose:
```
const completionStatus = (status: boolean = false): TransactionStatusEnum => {
      if (transaction.id === 0) {
        return TransactionStatusEnum.NOT_STARTED;
      }
      return transaction.block && status ? TransactionStatusEnum.COMPLETED : TransactionStatusEnum.IN_PROGRESS;
    };
```

outcome:
![Screenshot 2021-05-03 at 6 20 25 PM](https://user-images.githubusercontent.com/5408605/116897136-994ee880-ac3d-11eb-96fc-05a3c016bf88.png)

Code I propose:
```
const completionStatus = (status: boolean = false): TransactionStatusEnum => {
      if (transaction.id === 0) {
        return TransactionStatusEnum.NOT_STARTED;
      }
      if (!status) {
        return transaction.block ? TransactionStatusEnum.COMPLETED : TransactionStatusEnum.IN_PROGRESS;
      }
      return transaction.block && status ? TransactionStatusEnum.COMPLETED : TransactionStatusEnum.IN_PROGRESS;
    };
 ```
 
Outcome:
![Screenshot 2021-05-03 at 6 27 04 PM](https://user-images.githubusercontent.com/5408605/116897272-bf748880-ac3d-11eb-93a5-233610270c74.png)

Look at the 2nd line (" Rialto: Include message in block") - that is why I say that status matters.

Reply:
> Code you propose:

Please try this:
```js
 return transaction.block ? TransactionStatusEnum.COMPLETED : TransactionStatusEnum.IN_PROGRESS;
```

Obviously if you do `transaction.block && status` and `status == false` we will never get into completed state. I'm still not convinced that this is correct. If I'm right with your code all should steps should light up as completed as soon as `transaction.block` is set and I don't think that's expected behavior.

Reply:
Actually went through a trouble of running your code and it's definitely not correct cause it all turns into `COMPLETED` state right after the transaction is included as I suspected. Requesting changes then.